docs/tutorials/projects/bedtime-songs-bot.md

```diff
@@ -0,0 +1,555 @@
+
+Read through our guide to [training a new model](/manage/ml/train-model/) for more information.
+
+## Use your ML Model to sing songs to your kids
+
+### Configure your webcam to act as a shape classifier
+
+[Deploy the model](/services/ml/) to the robot and configure a [vision service](/services/vision/) of model `mlmodel` to use this model.
+
+Name your vision service `"shape-classifier"`.
+If you use a different name, adapt the code in the later steps of this tutorial to use the name you give your service.
+
+{{< tabs >}}
+{{% tab name="Raw JSON" %}}
+
+At this point, the full **Raw JSON** configuration of your robot should look like the following:
+
+``` json {class="line-numbers linkable-line-numbers"}
+{
+ "packages": [
+   {
+     "name": "shapes",
+     "version": "latest",
+     "package": "20055b44-c8a7-4bc5-ad93-86900ee9735a/shapes"
+   }
+ ],
+ "services": [
+   {
+     "name": "shape-classifier-model",
+     "type": "mlmodel",
+     "model": "tflite_cpu",
+     "attributes": {
+       "model_path": "${packages.shapes}/shapes.tflite",
+       "label_path": "${packages.shapes}/labels.txt",
+       "num_threads": 1
+     }
+   },
+   {
+     "name": "shape-classifier",
+     "type": "vision",
+     "model": "mlmodel",
+     "attributes": {
+       "mlmodel_name": "shape-classifier-model"
+     }
+   }
+ ],
+ "components": [
+   {
+     "name": "cam",
+     "type": "camera",
+     "model": "webcam",
+     "attributes": {},
+     "depends_on": []
+   }
+ ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Record bedtime songs
+
+Now, capture the audio of the songs you want your bot to play.
+
+- Record or download the audio files you want to use to your computer in <file>.mp3</file> format.
+- Make the names of the files match the classifier tags you used: for example, <file>square.mp3</file>.
+- Navigate to a directory where you want to store your SDK code.
+Save your audio files inside of this directory.
+
+The audio files I used are available to download on [GitHub](https://github.com/viam-labs/bedtime-songs-bot).
+
+### Program your bedtime-songs bot
+
+Now, use Viam's Go SDK to program your robot so that if your webcam "sees" (captures image data containing) a toy, the robot knows to play a particular song through its computer's speakers.
+
+Follow these instructions to start working on your Go control code:
+
+1. Navigate to your robot's page in [the Viam app](https://app.viam.com), and click on the **Code sample** tab.
+2. Select **Go** as the language.
+3. Click **Copy** to copy the generated code sample, which establishes a connection with your robot when run.
+
+   {{% snippet "show-secret.md" %}}
+
+4. Open your terminal.
+Navigate to the directory where you want to store your code.
+Paste this code sample into a new file named <file>play-songs.go</file>, and save it.
+
+For example, run the following commands on your Macbook to create and open the file:
+
+``` shell
```

Comment:
```suggestion
```sh {class="command-line" data-prompt="$"}
```

Reply:
done thank you for inline!